lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3InputStream.java

```diff
@@ -40,18 +40,20 @@ final class S3InputStream
     private final S3Client client;
     private final GetObjectRequest request;
     private final Long length;
+    private final S3Context context;
 
     private boolean closed;
     private ResponseInputStream<GetObjectResponse> in;
     private long streamPosition;
     private long nextReadPosition;
 
-    public S3InputStream(Location location, S3Client client, GetObjectRequest request, Long length)
+    public S3InputStream(Location location, S3Client client, GetObjectRequest request, Long length, S3Context context)
```

Comment:
Instead of having it as a part of `S3InputStream` - Can we have it as a part of `S3InputFile#newGetObjectRequest` ?